configs/terraform/environments/dev/artifact-registry/variables.tf

```
@@ -12,45 +12,18 @@ variable "gcp_project_id" {
   description = "Google Cloud project to create resources."
 }
 
-variable "artifact_registry_names" {
-  type        = list(string)
-  description = "Artifact Registry names"
-  default     = ["ocim", "internal"]
-}
-variable "artifact_registry_owner" {
-  type        = string
-  description = "Owner inside SAP"
-}
-
-variable "artifact_registry_module" {
-  type        = string
-  description = "Module name"
-}
-
-variable "artifact_registry_type" {
-  type        = string
-  description = "Environment for the resources"
-}
-
-variable "artifact_registry_serviceaccount" {
-  type        = string
-  description = "Service Account"
-}
-
-variable "artifact_registry_multi_region" {
-  type        = bool
-  description = "Is Location type Multi-region?"
-  default     = true
-}
-
-variable "artifact_registry_primary_area" {
-  type        = string
-  description = "Location type Multi-region"
-  default     = "europe"
+variable "artifact_registry_list" {
+  description = "Artifact Registry related data set"
+  type = list(object({
+    name                   = string
+    owner                  = string
+    type                   = string
+    writer_serviceaccount  = optional(string, "")
+    reader_serviceaccounts = list(string)
+    primary_area           = optional(string, "europe")
+    multi_region           = optional(bool, true)
+    public                 = optional(bool, false)
+    immutable              = optional(bool, false)
+  }))
 }
 
```

A variable declaration for map of objects.

```suggestion
variable "artifact_registries" {
  description = "Artifact Registry related data set"
  type = map(object({
    name                   = string
    owner                  = string
    type                   = string
    writer_serviceaccount  = optional(string, "")
    reader_serviceaccounts = list(string)
    primary_area           = optional(string, "europe")
    multi_region           = optional(bool, true)
    public                 = optional(bool, false)
    immutable              = optional(bool, false)
  }))
}

```